src/daft-json/src/read.rs

```diff
@@ -418,16 +418,15 @@ fn parse_into_column_array_chunk_stream(
                 let result = (move || {
                     // TODO(Clark): Switch to streaming parse + array construction?
                     let parsed = records
-                        .iter()
+                        .iter_mut()
                         .map(|unparsed_record| {
-                            json_deserializer::parse(unparsed_record.as_bytes()).map_err(|e| {
-                                super::Error::JsonDeserializationError {
+                            crate::deserializer::to_value(unsafe { unparsed_record.as_bytes_mut() })
```

Comment:
I'm assuming that the safety contract here for `as_bytes_mut()`, i.e. that the post-deserialization content of the slice is valid UTF8 before the parsed record is used, is trivially satisfied by the deserializer, + `records` is never used after parsing?

Reply:
yeah exactly. simd_json parser is "destructive" in nature, so as long as you don't attempt to reuse the original data after parsing, this is generally safe